website/source/docs/connect/proxies.html.md

```diff
@@ -54,6 +53,21 @@ Both of these default configurations help prevent arbitrary process
 execution or privilege escalation. This behavior can be configured
 [per-agent](/docs/agent/options.html#connect_proxy).
 
+### Lifecycle
+
+The Consul agent starts managed proxies on demand and supervises them,
+restarting them if they crash. The lifecycle of the proxy process is decoupled
+from the agent so if the agent crashes or is restarted for an upgrade, the
+managed proxy instances will _not_ be stopped.
+
+Note that this behaviour while desirable in production might leave proxy
+processes running indefinitely if you manually stop the agent and clear it's
+data dir during testing.
```

Comment:
I would add another paragraph stating that the recommended way to cleanly stop all proxies AND the agent is to first de-register all services on the agent.

I still think this is something we can improve from a workflow perspective. Its kind of jank. :) But bigger issue than these docs changes.

Reply:
Ah yeah I did in the callout note below but I'll repeat it here.

Reply:
Added below...